### PR TITLE
update falcon 7b model xpu

### PR DESCRIPTION
**Pull request by qiuxin2012**
## Description

fuse rope in falcon 7b model xpu

### 1. Why the change?

Speedup falcon 7b's performance on xpu.

### 2. User API changes

new falcon modelling file

### CODE REVIEW

**Comment by yangw1234** [comment]
https://github.com/intel-analytics/llm.cpp/pull/117 should support q and k of different sizes.

**Comment by hkvision** [comment]
> [intel-analytics/llm.cpp#117](https://github.com/intel-analytics/llm.cpp/pull/117) should support q and k of different sizes.

The perf is not improved after supporting this?
Seems even a little bit slower, not sure if it is due to instability.

File: python/llm/example/CPU/HF-Transformers-AutoModels/Model/falcon/falcon-7b-instruct/modelling_RW.py [hunk 1]
```diff
@@ -294,7 +295,13 @@ def forward(
...
 
             seq_len = seq_len + seq_len_past
-            position_ids = torch.tensor([range(seq_len_past, seq_len)], dtype=torch.long).to(query_layer.device)
+            if (seq_len_past, seq_len) in self.p_dict:
+                #print("get cache, " + str(seq_len_past) + "  " + str(seq_len))
+                position_ids = self.p_dict[(seq_len_past, seq_len)]
+            else:
+                #print("create new" + str(seq_len_past) + "  " + str(seq_len))
+                position_ids = torch.tensor([range(seq_len_past, seq_len)], dtype=torch.long).to(query_layer.device)
+                self.p_dict[(seq_len_past, seq_len)] = position_ids
```
Comment (jason-dai) [inline python/llm/example/CPU/HF-Transformers-AutoModels/Model/falcon/falcon-7b-instruct/modelling_RW.py:304]: do we need to store every seq_len?

Comment (qiuxin2012) [reply python/llm/example/CPU/HF-Transformers-AutoModels/Model/falcon/falcon-7b-instruct/modelling_RW.py:304]: Yes, store every seq_len is better than `new every time` and `cache one, and assign value`.
The best option should be add a new kernel function to accept seq_len directly.

**Comment by jason-dai** [comment]
We should follow the implementation of llama2

**Comment by qiuxin2012** [comment]
> We should follow the implementation of llama2

See https://github.com/intel-analytics/BigDL/pull/9242, close this one.

**Pull request closed without merging**